R/methods-Module.R

```diff
@@ -10,127 +9,183 @@
 #' }
 #' @param tree whether to use tree as latent space. If TRUE, object should have
 #' a tree slot.
-#' @param number_top_genes hotspot argument for number of genes to consider
+#' @param number_top_genes Hotspot argument for number of genes to consider
 #' @param num_umi optional dataframe containing umi counts in first column for
 #'  barcodes
-#' @param min_gene_threshold minimum number of genes in hotspot module
+#' @param min_gene_threshold minimum number of genes in Hotspot module
 #' @param n_neighbors number of neighbors to consider in latent space
 #' @param autocorrelation_fdr threshold for significance for genes autocorr
 #' @param clustering_fdr threshold for significance for clustering modules
+#' @param nn_precomp precomputed neighbors, barcodes x neighbors
+#' @param wt_precomp precomputed weights, barcodes x neighbors
 #' 
 #' Populates the modData, HotspotModuleScores, ModuleSignatureEnrichment
 #' and HotspotObject slots of object, as well as recalculates signature scores
 #' for new modules.
-calcHotspotModules <- function(object, model="normal", tree=FALSE, 
-                               number_top_genes=1000,num_umi=NULL, 
+#' @return the modified Vision object
+#' 
+#' @export
+hsAnalyze <- function(object, model="normal", tree=FALSE, 
+                               number_top_genes=1000, num_umi=NULL, 
                                min_gene_threshold=20, n_neighbors=NULL,
                                autocorrelation_fdr=0.05, clustering_fdr=0.5, nn_precomp=NULL, wt_precomp=NULL) {
 
-    hotspot <- import("hotspot", convert=F)
+    # Init Hotspot
+    hs <- hsInit(object, model, tree, num_umi)
+    # Init Hotspot KNN
+    hs <- hsCreateKnnGraph(hs, object, n_neighbors=NULL, nn_precomp=NULL, wt_precomp=NULL)
```

Comment:
If we haven't gotten `nn_precomp` and `wt_precomp` to work here, let's remove this from the argument list (for clarity)

Reply:
I think we can remove it if it's not working when merging into master but is it ok to keep it for this pr into yr-cass?

Reply:
No, let's remove it from this PR because we'll want to just merge in this PR to master when we're ready.